src/utils/convertData.js

```diff
@@ -0,0 +1,54 @@
+import dayjs from 'dayjs';
+import isBetween from 'dayjs/plugin/isBetween';
+import config from '../constants/config';
+
+const { days } = config;
+dayjs.extend(isBetween);
+
+const getTime = (date) => {
+  const time = date.set('m', 0);
+  return time.format('H');
+};
+
+const groupByDay = (posts) => {
+  const results = {};
+  // eslint-disable-next-line no-restricted-syntax
+  for (const post of posts) {
+    const date = dayjs.unix(post.created_utc);
+    const day = days[date.get('d')];
+    const time = getTime(date);
+
+    if (!results[day] || !results[day][time]) {
+      results[day] = { ...results[day], [time]: [] };
```

Comment:
This is not very performant since you clone the whole results object 7*24 times. Rather assign directly `results[day][time] = [];`

Reply:
the problem if I use an array, in this case, I can't  set [day][hour]

Reply:
Sure, if day is a number you can keep the same exact syntax